Convert project from Maven to Gradle

garyp opened the pull request:
Also had to:

- Update minimum Java version from 8 to 11 since that's required by Dropwizard 4
- Update the javax.validation imports to use the new jakarta package name

rahulreddym commented on buildSrc/src/main/kotlin/pro.streem.java-conventions.gradle.kts:
```diff
@@ -0,0 +1,27 @@
+}
+
+group = "pro.streem"
+version = "2.0.1-SNAPSHOT"
```
Shouldn't this be just `2.0.1`?

garyp replied:
Normally the version is "SNAPSHOT" as development happens. With every merge to `master`, a new snapshot build gets automatically published to the Sonatype Snapshot repository. When we're ready to do a release, we change the version from `2.0.1-SNAPSHOT` to `2.0.1` and tag it in git. Then that gets automatically published as a release version to Maven Central. Afterwards, we change the version from `2.0.1` to `2.0.2-SNAPSHOT` so that future merges to `master` will again cause snapshot builds to be published.

I'm sure there's a way to automate this so that the publishing process just automatically appends the `-SNAPSHOT` to the version when publishing from `master` (vs publishing from a git tag). But that's not how our GHA workflows are setup currently. So I was going for lowest effort, which was to copy the existing workflows from our other projects.

rahulreddym replied:
Makes sense now

rahulreddym commented on dropwizard-metrics-datadog/src/main/java/io/dropwizard/metrics/DatadogReporterFactory.java:
I got javax.validation not found error, but it went away when I added the jakarta.validation dependency, so I left it as-is. No action here but fyi

garyp replied:
Hmm, I was still getting package not found errors even with the `jakarta.validation` dependency you added. I'm not sure why, but when I looked at the `jakarta.validation` package contents, it didn't include the `javax.validation` classes. I suspect maybe we're building using different versions of Java, and newer Java versions don't package the `javax.validation` classes anymore?

rahulreddym reviewed (approved):
LGTM

garyp committed:
Convert project from Maven to Gradle

Also had to:

- Update minimum Java version from 8 to 11 since that's required by
  Dropwizard 4
- Update the javax.validation imports to use the new jakarta package
  name

garyp committed:
Add GitHub Actions workflows, and change version to -SNAPSHOT

garyp committed:
Fix DatadogReporterFactoryTest

The META-INF file for `DatadogReporterFactory` was named incorrectly,
causing it to not be discoverable.

garyp committed:
Fix remaining unit test failures
